File: generate-icons.js
#!/usr/bin/env node
/**
 * Generate Tauri app icons from a source image.
 * Requires: npm install sharp
 */

const fs = require('fs');
const path = require('path');

const ICONS_DIR = path.join(__dirname, 'src-tauri', 'icons');
const SOURCE_IMAGE = path.join(ICONS_DIR, 'source-icon.png');

async function generateIcons() {
  let sharp;
  try {
    sharp = require('sharp');
  } catch (err) {
    console.error('Error: sharp is required. Install it with:');
    console.error('  npm install sharp');
    process.exit(1);
  }

  if (!fs.existsSync(SOURCE_IMAGE)) {
    console.error(`Error: Source image not found: ${SOURCE_IMAGE}`);
    process.exit(1);
  }

  console.log('Generating Tauri app icons...');
  console.log(`Source: ${SOURCE_IMAGE}\n`);

  // Generate PNG icons
  const pngSizes = [
    { size: 32, filename: '32x32.png' },
    { size: 128, filename: '128x128.png' },
    { size: 256, filename: '128x128@2x.png' },
  ];

  console.log('Generating PNG icons...');
  for (const { size, filename } of pngSizes) {
    const outputPath = path.join(ICONS_DIR, filename);
    await sharp(SOURCE_IMAGE)
      .resize(size, size, { fit: 'contain', background: { r: 255, g: 255, b: 255, alpha: 0 } })
      .png()
      .toFile(outputPath);
    console.log(`  ✓ Created ${filename} (${size}x${size})`);
  }

  // Generate ICO file (Windows) - create PNG for conversion
  console.log('\nPreparing Windows ICO file...');
  const icoPng = path.join(ICONS_DIR, 'icon-256-for-ico.png');
  await sharp(SOURCE_IMAGE)
    .resize(256, 256, { fit: 'contain', background: { r: 255, g: 255, b: 255, alpha: 0 } })
    .png()
    .toFile(icoPng);
  console.log(`  ✓ Created ${path.basename(icoPng)} for ICO conversion`);
  console.log('  → Convert to ICO: https://convertio.co/png-ico/');
  console.log('     Upload the PNG above and download as icon.ico');

  console.log('\n' + '='.repeat(60));
  console.log('ICNS file (macOS) generation:');
  console.log('='.repeat(60));
  console.log('ICNS files require macOS-specific tools.');
  console.log('\nOption 1: Use online converter');
  console.log('  - Go to: https://cloudconvert.com/png-to-icns');
  console.log('  - Upload: src-tauri/icons/128x128@2x.png (256x256)');
  console.log('  - Download and save as: src-tauri/icons/icon.icns');
  console.log('='.repeat(60));

  console.log('\n✓ Icon generation complete!');
  console.log('\nNext steps:');
  console.log('1. If ICO generation failed, convert icon-256.png to icon.ico online');
  console.log('2. Generate icon.icns using one of the methods above');
  console.log('3. Rebuild your Tauri app: npm run tauri build');
}

generateIcons().catch(err => {
  console.error('Error:', err.message);
  process.exit(1);
});

File: generate-icons.py
#!/usr/bin/env python3
"""
Generate Tauri app icons from a source image.
Requires: pip install Pillow
"""

import sys
from pathlib import Path

try:
    from PIL import Image
except ImportError:
    print("Error: Pillow is required. Install it with: pip install Pillow")
    sys.exit(1)

ICONS_DIR = Path("src-tauri/icons")
SOURCE_IMAGE = ICONS_DIR / "source-icon.png"

def generate_png_icons():
    """Generate PNG icons at required sizes."""
    print("Loading source image...")
    img = Image.open(SOURCE_IMAGE)
    
    # Convert to RGBA if needed
    if img.mode != 'RGBA':
        img = img.convert('RGBA')
    
    sizes = [
        (32, 32, "32x32.png"),
        (128, 128, "128x128.png"),
        (256, 256, "128x128@2x.png"),
    ]
    
    print("Generating PNG icons...")
    for width, height, filename in sizes:
        resized = img.resize((width, height), Image.Resampling.LANCZOS)
        output_path = ICONS_DIR / filename
        resized.save(output_path, "PNG")
        print(f"  ✓ Created {filename} ({width}x{height})")

def generate_ico():
    """Generate Windows ICO file with multiple sizes."""
    print("Generating Windows ICO file...")
    img = Image.open(SOURCE_IMAGE)
    
    if img.mode != 'RGBA':
        img = img.convert('RGBA')
    
    # ICO format requires multiple sizes
    sizes = [16, 32, 48, 256]
    icons = []
    
    for size in sizes:
        resized = img.resize((size, size), Image.Resampling.LANCZOS)
        icons.append(resized)
    
    output_path = ICONS_DIR / "icon.ico"
    icons[0].save(
        output_path,
        format='ICO',
        sizes=[(s, s) for s in sizes]
    )
    print(f"  ✓ Created icon.ico with sizes: {sizes}")

def generate_icns_info():
    """Print instructions for generating ICNS file."""
    print("\n" + "="*60)
    print("ICNS file (macOS) generation:")
    print("="*60)
    print("ICNS files are macOS-specific and harder to generate on Linux.")
    print("\nOption 1: Use online converter")
    print("  - Go to: https://cloudconvert.com/png-to-icns")
    print("  - Upload: src-tauri/icons/128x128@2x.png (256x256)")
    print("  - Download and save as: src-tauri/icons/icon.icns")
    print("\nOption 2: Use macOS (if available)")
    print("  - Use 'iconutil' command on macOS")
    print("  - Or use online tools like: https://convertio.co/png-icns/")
    print("="*60)

def main():
    if not SOURCE_IMAGE.exists():
        print(f"Error: Source image not found: {SOURCE_IMAGE}")
        sys.exit(1)
    
    print("Generating Tauri app icons...")
    print(f"Source: {SOURCE_IMAGE}\n")
    
    generate_png_icons()
    print()
    generate_ico()
    print()
    generate_icns_info()
    
    print("\n✓ Icon generation complete!")
    print("\nNext steps:")
    print("1. Generate icon.icns using one of the methods above")
    print("2. Rebuild your Tauri app to see the new icons")

if __name__ == "__main__":
    main()
